packages/canon/src/components/Button/Button.tsx

```diff
@@ -26,34 +25,46 @@ import type { IconNames } from '../Icon/types';
  */
 export interface ButtonProps {
   size?: 'small' | 'medium';
-  variant?: 'primary' | 'secondary';
+  variant?: 'primary' | 'secondary' | 'tertiary';
   children: React.ReactNode;
   disabled?: boolean;
   iconStart?: IconNames;
   iconEnd?: IconNames;
 }
 
 /** @public */
-export const Button = ({
-  size = 'medium',
-  variant = 'primary',
-  children,
-  disabled,
-  iconStart,
-  iconEnd,
-  ...props
-}: ButtonProps) => {
-  return (
-    <button
-      {...props}
-      disabled={disabled}
-      className={button({ size, variant, disabled })}
-    >
-      {iconStart && <Icon name={iconStart} />}
-      {children}
-      {iconEnd && <Icon name={iconEnd} />}
-    </button>
-  );
-};
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      size = 'medium',
+      variant = 'primary',
+      children,
+      disabled,
+      iconStart,
+      iconEnd,
+      ...props
+    }: ButtonProps,
+    ref,
+  ) => {
+    return (
+      <button
+        {...props}
+        ref={ref}
+        disabled={disabled}
+        className={`button ${variant} ${size}`}
+      >
+        <span
+          className={['button-content', iconStart && iconEnd ? 'icon-both' : '']
+            .filter(Boolean)
+            .join(' ')}
+        >
+          {iconStart && <Icon name={iconStart} />}
+          {children}
+        </span>
+        {iconEnd && <Icon name={iconEnd} />}
```

Comment:
Hmm, not a big fan of the extra API surface that this exposes, but not gonna worry took much yet given the button is somewhat special